vtm/src/org/oscim/renderer/ExtrusionRenderer.java

```diff
@@ -319,7 +347,8 @@ public void render(GLViewport v) {
             }
 
             /* just a temporary reference! */
-            ebs[i] = null;
+            /* But for shadows we use them multiple times */
+            //ebs[i] = null;
```

Comment:
Can we keep here the null set as before, when not using shadows?

Reply:
That needs an extra local variable. E.g. such as "mMultipleRenders". But is this really needed?

Reply:
In any case `mShader` is available and could use a check like in [line:373](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/renderer/ExtrusionRenderer.java#L373)?

Reply:
Works, but not the cleanest solution :)

Reply:
I tested with Android Studio Profiler and I couldn't notice any remarkable memory regression when holding `ExtrusionBuckets` in  `ebs`. If really needed, we should null all not used entries of `mExtrusionBucketSet` in `BuildingRenderer.update` [after new `ExtrusionBuckets` are assigned](https://github.com/mapsforge/vtm/blob/6b0e521c755888a6992f38f0b841b518955a859b/vtm/src/org/oscim/layers/tile/buildings/BuildingRenderer.java#L189) (use `activeTile` count to null the rest).

```
for(int i = activeTiles; i < mExtrusionBucketSet.length && mExtrusionBucketSet[i] != null; i++) {
    mExtrusionBucketSet[i] = null;
}
```

But may it's not even needed or even costs more performance to explicitly nulling them, e.g. tags of `MapElement` aren't nulled either.